server/pulp/server/managers/content/query.py

```diff
@@ -356,13 +345,8 @@ def _build_multi_keys_spec(content_type, unit_keys_dicts):
         missing_keys = key_fields_set.difference(keys_dict_set)
         if missing_keys:
             keys_errors.append(missing_keys_msg % {'a': ','.join(missing_keys), 'b': ','.join(key_fields)})
-        if extra_keys or missing_keys:
-            continue
-        # validation passed, store the keys and values in the template
-        for k, v in keys_dict.items():
-            spec_template[k].add(v)
     if keys_errors:
         value_error_msg = '\n'.join(keys_errors)
         raise ValueError(value_error_msg)
-    spec = dict([(k, {'$in': list(v)}) for k, v in spec_template.items()])
+    spec = {'$or': unit_keys_dicts}
```

Comment:
I like that in addition the being more correct, this is a lot easier to understand.